olive/passes/onnx/nvmo_quantization.py

```diff
@@ -0,0 +1,118 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------
+from copy import deepcopy
+from pathlib import Path
+from typing import Any, Callable, Dict, Union
+
+from olive.cache import get_local_path_from_root
+from olive.data.config import DataConfig
+from olive.hardware.accelerator import AcceleratorSpec
+from olive.model import OliveModelHandler
+from olive.model.utils import resolve_onnx_path
+from olive.passes import Pass
+from olive.passes.onnx.common import model_proto_to_olive_model
+from olive.passes.pass_config import ParamCategory, PassConfigParam
+from olive.resource_path import OLIVE_RESOURCE_ANNOTATIONS
+from olive.strategy.search_parameter import Categorical
+
+# TODO(@riyadshairi979): add use of `approach` and `algorithm` config
+# TODO(@riyadshairi979): add validation, but on what EP?
+# TODO(@riyadshairi979): add unit tests
+
+# static quantization specific config
+_dataloader_config = {
+    "data_dir": PassConfigParam(
+        type_=OLIVE_RESOURCE_ANNOTATIONS,
+        category=ParamCategory.DATA,
+        description="""
+            Path to the directory containing the dataset.
+            For local data, it is required if quant_mode is 'static' and dataloader_func is provided.
+        """,
+    ),
+    "batch_size": PassConfigParam(
+        type_=int,
+        default_value=1,
+        description="""
+            Batch size for calibration, only used if dataloader_func is provided.
+        """,
+    ),
+    # TODO(trajep): remove this option once we have a data config ready
+    "dataloader_func": PassConfigParam(
+        type_=Union[Callable, str],
+        category=ParamCategory.OBJECT,
+        description="""
+            Function/function name to generate dataloader for calibration,
+            required if quant_mode is 'static' and data_config is None.
+        """,
+    ),
+    "dataloader_func_kwargs": PassConfigParam(
+        type_=Dict[str, Any],
+        description="Keyword arguments for dataloader_func.",
+    ),
+    "data_config": PassConfigParam(
+        type_=Union[DataConfig, Dict],
+        description="""
+            Data config for calibration, required if quant_mode is 'static' and
+            dataloader_func is None.
+        """,
+    ),
+}
+
+
+class NVModelOptQuantization(Pass):
+
+    # set this to True if the pass has parameters that are functions or objects and the user script is required
+    # to import the module containing the function or object
+    _requires_user_script: bool = True
+
+    @classmethod
+    def _default_config(cls, accelerator_spec: AcceleratorSpec) -> Dict[str, PassConfigParam]:
+        config = {
+            "approach": PassConfigParam(
+                type_=str,
+                default_value="static",
```

Comment:
this default value doesn't match the searchable values. Also, the parameter might be better named as "precision"?

Reply:
Oh it looks like only int4 is support right now. If so, this field can be removed as well.

Reply:
"algoritm is not used as well